File: Models/Driver.cs
using System;
using System.Text.Json;

namespace GTrack.Models
{
    public class Driver : IdentificableItem
    {
        public string uid { get; set; }

        public string email { get; set; }

        public string name { get; set; }

        public string surname { get; set; }

        public string phoneNumber { get; set; }

        public override string ToString()
        {
            return JsonSerializer.Serialize<Driver>(this);
        }
    }
}

File: ClientApp/src/app/modules/content-modules/drivers/models/driver.ts

export interface IDriver {
    uid: string,
    email: string, 
    name: string, 
    surname: string
    phoneNumber: string
}

export class Driver implements IDriver{
    uid: string = ""
    email: string = ""
    name: string = ""
    surname: string = ""
    phoneNumber: string = ""

    get _compositeName(): string{
        return this.name + " " + this.surname;
    }

    constructor(d : Driver){
        this.uid = d.uid;
        this.email = d.email;
        this.name = d.name;
        this.surname = d.surname;
        this.phoneNumber = d.phoneNumber;
    }
    
}

